Examples/Detectors/GeoModelDetector/src/GeoModelMuonMockupBuilder.cpp

```diff
@@ -0,0 +1,163 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/GeoModelDetector/GeoModelMuonMockupBuilder.hpp"
+
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/Geometry/Blueprint.hpp"
+#include "Acts/Geometry/ContainerBlueprintNode.hpp"
+#include "Acts/Geometry/CylinderVolumeBounds.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/TrackingVolume.hpp"
+#include "Acts/Geometry/TrapezoidVolumeBounds.hpp"
+#include "Acts/Geometry/VolumeAttachmentStrategy.hpp"
+#include "Acts/Utilities/AxisDefinitions.hpp"
+#include "Acts/Utilities/MathHelpers.hpp"
+
+#include "GeoModelKernel/throwExcept.h"
+
+using namespace Acts::UnitLiterals;
+
+namespace ActsExamples {
+
+GeoModelMuonMockupBuilder::GeoModelMuonMockupBuilder(
+    const Config& cfg, std::unique_ptr<const Acts::Logger> logger)
+    : m_cfg(cfg), m_logger(std::move(logger)) {}
+
+std::unique_ptr<const Acts::TrackingGeometry>
+GeoModelMuonMockupBuilder::trackingGeometry(
+    const Acts::GeometryContext& gctx) const {
+  ConvertedVolList_t boundingBoxes = m_cfg.volumeBoxFPVs;
+
+  // Blue print construction for the tracking geometry
+  Acts::Experimental::Blueprint::Config bpCfg;
+  bpCfg.envelope[Acts::AxisDirection::AxisZ] = {20_mm, 20_mm};
+  bpCfg.envelope[Acts::AxisDirection::AxisR] = {5008.87000_mm, 12_mm};
+  Acts::Experimental::Blueprint root{bpCfg};
+  auto& cyl = root.addCylinderContainer("MuonMockupBarrelContainer",
+                                        Acts::AxisDirection::AxisR);
+  cyl.setAttachmentStrategy(Acts::VolumeAttachmentStrategy::Gap);
+  cyl.setResizeStrategy(Acts::VolumeResizeStrategy::Gap);
+
+  if (boundingBoxes.empty()) {
+    THROW_EXCEPTION(
```

Comment:
I would prefer a standard `throw`. It's not really necessary to get the code location for this, if the message is informative.